components/page-templates/B2B/AccountHierarchyTemplate/AccountHierarchyTemplate.tsx

```diff
@@ -23,32 +43,132 @@ const AccountHierarchyTemplate = () => {
+    setAccountHierarchy({
+      accounts: b2BAccountHierarchy?.accounts,
+      hierarchy,
+    })
+  }, [b2BAccountHierarchy, data])
 
   const onAccountTitleClick = () => {
     router.push('/my-account')
   }
 
-  const handleFormSubmit = async (formValues: CreateCustomerB2bAccountParams) => {
+  const handleAddAccountFormSubmit = async (formValues: CreateCustomerB2bAccountParams) => {
     const variables = buildCreateCustomerB2bAccountParams({
       ...formValues,
-      parentAccount: user as B2BAccount,
     })
     const createCustomerB2BAccount = await createCustomerB2bAccount.mutateAsync({
       ...variables,
     })
     if (createCustomerB2BAccount) closeModal()
   }
 
+  const handleEditAccountFormSubmit = async (
+    formValues: CreateCustomerB2bAccountParams,
+    account: B2BAccount
+  ) => {
+    console.log(formValues, account)
+    const variables = buildUpdateCustomerB2bAccountParams(formValues, account)
+    const updateCustomerB2BAccount = await updateCustomerB2bAccount.mutateAsync({
+      ...variables,
+    })
+    if (updateCustomerB2BAccount) closeModal()
+  }
+
+  const handleAddAccount = ({ isAddingAccountToChild, accounts }: AddChildAccountProps) => {
+    showModal({
+      Component: AccountHierarchyAddFormDialog,
+      props: {
+        accounts,
+        isAddingAccountToChild,
+        primaryButtonText: t('create-account'),
+        onSave: (formValues: CreateCustomerB2bAccountParams) =>
+          handleAddAccountFormSubmit(formValues),
+        onClose: () => closeModal(),
+      },
+    })
+  }
+
+  const handleEditAccount = ({ accounts, accountToEdit }: EditChildAccountProps) => {
+    showModal({
+      Component: AccountHierarchyEditFormDialog,
+      props: {
+        accounts,
+        accountToEdit,
+        primaryButtonText: t('save'),
+        formTitle: t('edit-child-account'),
+        onSave: (formValues: CreateCustomerB2bAccountParams) =>
+          handleEditAccountFormSubmit(formValues, accountToEdit),
+        onClose: () => closeModal(),
+      },
+    })
+  }
+
+  const handleViewAccount = (b2BAccount: B2BAccount) => {
+    showModal({
+      Component: ViewAccountDetailsDialog,
+      props: {
+        b2BAccount,
+        onClose: () => closeModal(),
+      },
+    })
+  }
+
+  const handleSwapAccount = (b2BAccount: B2BAccount) => {
+    showModal({
+      Component: ConfirmationDialog,
+      props: {
+        contentText: t('swap-account-hierarchy-text'),
+        primaryButtonText: t('yes-move-anyway'),
+        title: t('swap-account-hierarchy'),
+        onConfirm: () => console.log('account swapped'),
+      },
+    })
+  }
+
+  const handleDeleteAccount = (b2BAccount: B2BAccount) => {
+    showModal({
+      Component: ConfirmationDialog,
+      props: {
+        contentText: t('delete-account-confirm-message'),
+        primaryButtonText: t('yes-remove'),
+        title: t('confirmation'),
+        onConfirm: () => console.log('account deleted'),
+      },
+    })
+  }
```

Comment:
As there is no Delete functionality change  handleDeleteAccount to  handleDisableAccount(b2BAccountId: string)

Reply:
Done